client/src/key_management/key_manager.rs

```diff
@@ -17,18 +16,19 @@ pub struct KeyManager {
 
 impl KeyManager {
     pub fn new() -> KeyManager {
-        let mut key_incrementer_map = HashMap::new();
-        key_incrementer_map.insert(PreKey::Signed, 0);
-        key_incrementer_map.insert(PreKey::Kyber, 0);
-        key_incrementer_map.insert(PreKey::OneTime, 0);
         Self {
-            key_incrementer_map,
+            key_incrementer_map: HashMap::from([
+                (PreKey::Signed, 0u32),
+                (PreKey::Kyber, 0u32),
+                (PreKey::OneTime, 0u32),
+            ]),
         }
     }
 
     fn get_new_key_id(&mut self, key_type: &PreKey) -> u32 {
-        let id = self.key_incrementer_map.get_mut(key_type).unwrap().deref() + 1u32;
-        id - 1
+        let id = self.key_incrementer_map.get(key_type).unwrap().clone();
```

Comment:
unwrap bad

Reply:
Nope we know that this key exist in that it was created with the object